main.tf

```diff
@@ -17,126 +21,194 @@ resource "aws_s3_bucket" "default" {
   #bridgecrew:skip=BC_AWS_S3_16:Skipping `Ensure S3 bucket versioning is enabled` because dynamic blocks are not supported by checkov
   #bridgecrew:skip=BC_AWS_S3_14:Skipping `Ensure all data stored in the S3 bucket is securely encrypted at rest` because variables are not understood
   #bridgecrew:skip=BC_AWS_GENERAL_56:Skipping `Ensure that S3 buckets are encrypted with KMS by default` because we do not have good defaults
-  count               = local.enabled ? 1 : 0
-  bucket              = local.bucket_name
-  acl                 = try(length(var.grants), 0) == 0 ? var.acl : null
-  force_destroy       = var.force_destroy
-  tags                = module.this.tags
-  acceleration_status = var.transfer_acceleration_enabled ? "Enabled" : null
-
-  dynamic "versioning" {
-    for_each = var.versioning_enabled ? [true] : []
+  count         = local.enabled ? 1 : 0
+  bucket        = local.bucket_name
+  force_destroy = var.force_destroy
+
+  dynamic "object_lock_configuration" {
+    for_each = var.object_lock_configuration != null ? [1] : []
+
     content {
-      enabled = true
+      object_lock_enabled = "Enabled"
     }
   }
 
-  dynamic "lifecycle_rule" {
+  tags = module.this.tags
+}
+
+resource "aws_s3_bucket_accelerate_configuration" "default" {
+  count  = local.transfer_acceleration_enabled ? 1 : 0
+  bucket = join("", aws_s3_bucket.default.*.id)
+  status = "Enabled"
```

Comment:
Might want to allow `suspended` here at some point, but that can be down the road